### PR TITLE
Add Retry Logic + Bump Version

### PR DESCRIPTION
**Pull request by dwallace0723**
This PR adds retry logic to the Pagerduty tap. It uses the [backoff](https://github.com/litl/backoff) decorator to implement a fibonaccial backoff decay upon a fatal response status code. It also implements a max retry time of 120 seconds and adds retry logging to our existing handler.

### CODE REVIEW

File: tests/test_client.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import pytest
+import requests
+
+from tap_pagerduty.streams import is_fatal_code
+
+
+@pytest.mark.parametrize('status_code', [400, 401, 403, 404,
```
Comment (bobzoller) [inline tests/test_client.py:9]: same comment as other PR (add a test for `Timeout` or `ConnectionError`)

**Review by bobzoller — CHANGES_REQUESTED** [review]

File: tests/test_client.py (hunk 1 above)
Comment (dwallace0723) [reply tests/test_client.py:9]: @bobzoller addressed with same decorator routing rationale as [#4](https://github.com/goodeggs/tap-wonolo/pull/4)

**Comment by bobzoller** [comment]
assume you'll update this one to mirror https://github.com/goodeggs/tap-wonolo/pull/4 now

**Review by bobzoller — APPROVED** [review]
awesome!

**Pull request merged**